src/parquet/column/reader.cc

```diff
@@ -37,32 +37,42 @@ ColumnReader::ColumnReader(const ColumnDescriptor* descr,
 
 template <int TYPE>
 void TypedColumnReader<TYPE>::ConfigureDictionary(const DictionaryPage* page) {
-  int encoding = static_cast<int>(Encoding::RLE_DICTIONARY);
+  Encoding::type encode_type = page->encoding();
+  if (encode_type == Encoding::PLAIN_DICTIONARY ||
+      encode_type == Encoding::PLAIN) {
+    encode_type = Encoding::RLE_DICTIONARY;
+  }
+
+  int encoding = static_cast<int>(encode_type);
 
   auto it = decoders_.find(encoding);
   if (it != decoders_.end()) {
     throw ParquetException("Column cannot have more than one dictionary.");
   }
 
-  PlainDecoder<TYPE> dictionary(descr_);
-  dictionary.SetData(page->num_values(), page->data(), page->size());
-
-  // The dictionary is fully decoded during DictionaryDecoder::Init, so the
-  // DictionaryPage buffer is no longer required after this step
-  //
-  // TODO(wesm): investigate whether this all-or-nothing decoding of the
-  // dictionary makes sense and whether performance can be improved
-
-  auto decoder = std::make_shared<DictionaryDecoder<TYPE> >(descr_);
-  decoder->SetDict(&dictionary);
+  if (encode_type == Encoding::RLE_DICTIONARY) {
```

Comment:
This is misleading -- we could check for `Encoding::PLAIN` or `Encoding::PLAIN_DICTIONARY` here -- however we _do_ want to store the decoder in `decoders_` under the `RLE_DICTIONARY` key